packages/core/src/chain.ts

```diff
@@ -66,21 +66,12 @@ async function setChain(options: {
       filter(x => x === null),
       map(() => false)
     )
-
     if (
       code === ProviderRpcErrorCode.CHAIN_NOT_ADDED ||
       code === ProviderRpcErrorCode.UNRECOGNIZED_CHAIN_ID
     ) {
       // chain has not been added to wallet
-      try {
-        await addNewChain(wallet.provider, chain)
-        await switchChain(wallet.provider, chainIdHex)
-        return true
-      } catch (error) {
-        // display notification to user to switch chain
-        switchChainModal$.next({ chain })
-        return firstValueFrom(switchChainModalClosed$)
-      }
+      chainNotInWallet(wallet, chain, switchChainModal$, chainIdHex)
```

Comment:
Do we need a `return` statement here?
```suggestion
      return chainNotInWallet(wallet, chain, switchChainModal$, chainIdHex)
```